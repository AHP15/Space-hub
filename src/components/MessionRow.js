import PropTypes from 'prop-types';

// eslint-disable-next-line object-curly-newline, no-unused-vars
const MessionRow = ({ id, missionName, description, changeBg, joined }) => {
  const handleJoin = () => {
  };

  return (
    <tr className="table_head" style={{ backgroundColor: changeBg ? 'rgba(211, 211, 211, 0.5)' : '' }}>
      <th>{missionName}</th>
      <td>{description}</td>
      <td className="btn_row">
        <button type="button">NOT A MEMBER</button>
      </td>
      <td className="btn_row">
        <button type="button" onClick={handleJoin}>Join Mission</button>
      </td>
    </tr>
  );
};

MessionRow.propTypes = {
  id: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  changeBg: PropTypes.bool.isRequired,
  joined: PropTypes.bool.isRequired,
};

export default MessionRow;
