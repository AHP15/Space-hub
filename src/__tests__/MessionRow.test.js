import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider, useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import MessionRow from '../components/MessionRow';
import Store from '../slices/store';

it('renders MessionRow correctly', () => {
  const tree = renderer.create(
    <Provider store={Store}>
      <MessionRow
        id={Math.random().toString()}
        missionName="mission name"
        description="mission description mission description mission description mission"
        changeBg={false}
        joined={false}
      />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

const CustomizedCompenent = ({ index }) => {
  const { messions } = useSelector((state) => state.missions);
  return (
    <div>
      <table>
        <tbody>
          <MessionRow
            id={messions[index].mission_id}
            missionName={messions[index].mission_name}
            description={messions[index].description}
            changeBg={false}
            joined={messions[index]?.joined}
          />
        </tbody>
      </table>
    </div>
  );
};

CustomizedCompenent.propTypes = {
  index: PropTypes.number.isRequired,
};

it('Join button works correctly', async () => {
  // Arrange
  render(<Provider store={Store}><CustomizedCompenent index={0} /></Provider>);

  // Act
  await userEvent.click(screen.getByText('Join Mission'));
  await screen.getByTitle('isMember');

  // Assertion
  expect(screen.getByTitle('isMember')).toHaveTextContent('Active Member');
});

it('Leave button works correctly', async () => {
  // Arrange
  render(<Provider store={Store}><CustomizedCompenent index={1} /></Provider>);

  // Act
  await userEvent.click(screen.getByText('Leave Mission'));
  await screen.getByTitle('isMember');

  // Assertion
  expect(screen.getByTitle('isMember')).toHaveTextContent('NOT A MEMBER');
});
