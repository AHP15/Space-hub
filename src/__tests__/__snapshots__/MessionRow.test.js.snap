// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MessionRow correctly 1`] = `
<tr
  className="table_head"
  style={
    Object {
      "backgroundColor": "",
    }
  }
>
  <th>
    mission name
  </th>
  <td>
    mission description mission description mission description mission
  </td>
  <td
    className="btn_row"
  >
    <button
      className="not_member"
      title="isMember"
      type="button"
    >
      NOT A MEMBER
    </button>
  </td>
  <td
    className="btn_row"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
